### ISSUE TITLE
Container runs as root but kustomize config includes runAsNonRoot

### ISSUE THREAD

**Issue by GJKrupa** [kubernetes-sigs/descheduler#532]
**What version of descheduler are you using?**

descheduler version: v0.20.0

**Does this issue reproduce with the latest release?**
Yes

**Which descheduler CLI options are you using?**
I'm using the default config from github.com/kubernetes-sigs/descheduler/kubernetes/cronjob?ref=master

**Please provide a copy of your descheduler policy config file**
I'm using the default config from github.com/kubernetes-sigs/descheduler/kubernetes/cronjob?ref=master

**What k8s version are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4+k3s1", GitCommit:"838a906ab5eba62ff529d6a3a746384eba810758", GitTreeState:"clean", BuildDate:"2021-02-22T19:49:35Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/arm64"}
</pre></details>

**What did you do?**
Installed into a Raspberry Pi k3s cluster by typing `kustomize build 'github.com/kubernetes-sigs/descheduler/kubernetes/cronjob?ref=master' | kubectl apply -f -`

**What did you expect to see?**

A descheduler job runs successfully within 2 minutes

**What did you see instead?**

The pod failed to start with CreateContainerConfigError due to the following:
> Error: container has runAsNonRoot and image will run as root (pod: "descheduler-cronjob-1616613840-2rbww_kube-system(87e096d0-d6a │
│ 2-4f55-9dec-30e9f502f96b)", container: descheduler)

I had to work around this by modifying the config to add runAsUser: 1000 and runAsGroup: 1000

**Comment by seanmalloy** [comment]
@GJKrupa the problem is k8s YAML manifests on the master branch are already updated for the descheduler v0.21.0 release.

This command should work for you:
```
kustomize build 'github.com/kubernetes-sigs/descheduler/kubernetes/cronjob?ref=release-1.20' | kubectl apply -f -
```

At a minimum we should probably update the docs, so this is more clear. There might also be a way to update the kustomize files to make this more seemly too.

/kind documentation

**Comment by damemi** [comment]
> the problem is k8s YAML manifests on the master branch are already updated for the descheduler v0.21.0 release.

yeah, master is almost always the in-progress code for the next release, so mixing it with an image from a previous release isn't guaranteed. The v0.21.0 release image will be published soon after the Kubernetes 1.21 release (around April 8). At which point you will be able to pull these manifests from the `release-1.21` branch

I think this is a duplicate of what was already discussed in https://github.com/kubernetes-sigs/descheduler/issues/466#issuecomment-781071548, so we can probably close it
/remove-kind bug

**Comment by seanmalloy** [comment]
@damemi I'll make a small tweak to the README in the kustomize section, so that this is less confusing.

/assign
